contribution_plan/gql/gql_mutations/contribution_plan_bundle_mutations.py

```diff
@@ -45,5 +58,41 @@ class ReplaceContributionPlanBundleMutation(BaseHistoryModelReplaceMutationMixin
     _mutation_module = "contribution_plan"
     _model = ContributionPlanBundle
 
+    @classmethod
+    def _mutate(cls, user, **data):
+        super()._mutate(user, **data)
+        # copy attached contribution plan bundles into new version
+        list_cpbd = ContributionPlanBundleDetails.objects.filter(
+            contribution_plan_bundle__id=data["uuid"],
+            is_deleted=False,
+        )
+        old_cpb = ContributionPlanBundle.objects.filter(id=data["uuid"], is_deleted=False).first()
+        new_cpb = ContributionPlanBundle.objects.filter(id=old_cpb.replacement_uuid, is_deleted=False).first()
+        if new_cpb:
+            for cpbd in list_cpbd:
+                cls._attach_contribution_plan_to_new_version_of_bundle(
+                    user,
+                    cpbd.contribution_plan.id,
+                    new_cpb.id,
+                    new_cpb.date_valid_from,
+                    new_cpb.date_valid_to
+                )
+
```

Comment:
I don't see the update on the 
OLD.date_valid_to  = NEW.date_valid_from if OLD.date_valid_to  > NEW.date_valid_from else OLD.date_valid_to

Reply:
ok. I forgot to consider this part. I will add this in this logic.

Reply:
fixed in commit fdf244be8a915f1af74f290e23a0836dba3104a4